src/test/java/crawlercommons/domains/EffectiveTldFinderTest.java

```diff
@@ -136,11 +140,53 @@ public final void testCoJpDomain() throws Exception {
     }
 
     @Test
+    public final void testAcZaDomain() throws Exception {
+        // .za isn't a TLD while .ac.za, .alt.za, .co.za, etc. are TLDs
+        // according to (as of 2017-08-03)
+        // https://www.zadna.org.za/content/page/domain-information/
+        EffectiveTLD etld = EffectiveTldFinder.getEffectiveTLD("blogs.uct.ac.za");
+        assertNotNull(etld);
+        assertEquals("ac.za", etld.getDomain());
+        String ad = null;
+        ad = EffectiveTldFinder.getAssignedDomain("blogs.uct.ac.za");
+        assertEquals("uct.ac.za", ad);
+    }
+
+    @Test
     public final void testUkDomain() throws Exception {
         String ad = null;
         ad = EffectiveTldFinder.getAssignedDomain("bbc.co.uk");
         assertEquals("bbc.co.uk", ad);
         ad = EffectiveTldFinder.getAssignedDomain("www.bbc.co.uk");
         assertEquals("bbc.co.uk", ad);
     }
+
+    @Test
+    public final void testMixedCaseHostNames() throws Exception {
+        String ad = null;
+        ad = EffectiveTldFinder.getAssignedDomain("www.PayPal.COM");
+        assertEquals("paypal.com", ad);
+    }
+
+    @Test
+    public final void testIDNDomain() throws Exception {
+        String ad = null;
+        ad = EffectiveTldFinder.getAssignedDomain("спб.бесплатныеобъявления.рф");
+        // assertEquals("бесплатныеобъявления.рф", ad); // TODO
+        ad = EffectiveTldFinder.getAssignedDomain("xn--90a1af.xn--80abbembcyvesfij3at4loa4ff.xn--p1ai");
+        assertEquals("xn--80abbembcyvesfij3at4loa4ff.xn--p1ai", ad);
+        // rare but possible mixed use of UTF-8 and Punycode
+        ad = EffectiveTldFinder.getAssignedDomain("xn--90a1af.бесплатныеобъявления.рф");
+        // assertEquals("xn--80abbembcyvesfij3at4loa4ff.xn--p1ai", ad); // TODO
```

Comment:
Add link to new issue for commented out lines?

Reply:
ok